addons/im_livechat/static/src/legacy/public_livechat_chatbot.js

```diff
@@ -179,8 +179,10 @@ const QWeb = core.qweb;
         });
 
         const welcomeMessagesIds = welcomeMessages.map(welcomeMessage => welcomeMessage._id);
-        this._messages = this._messages.filter((message) => {
-            !welcomeMessagesIds.includes(message._id);
+        this.messaging.livechatButtonView.update({
+            messages: this.messaging.livechatButtonView.messages.filter((message) => {
+                !welcomeMessagesIds.includes(message._id);
+            }),
```

Comment:
no return? does this filter ever worked? I feel like the result will always be an empty array, which is probably not what was intended

Reply:
Yes, I also saw this issue in the original code.
It's likely an error and the intend is to return... But fixing it now would be a change of behaviour, and it's likely unsafe (e.g. due to some badly written tests or tours that rely on this buggy behaviour, and could potentially break with "fixing" it).

So I'm aware of this, but I'm much prefer to keep it that way, while it's not fully moved to models.
Fixing behaviour is far easier in models than doing it now. Thanks!